src/pilot/assign_role.py

```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
     return raid_controller_ids
 
 
-def define_controller_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
```

Comment:
Is it cleaner to only pass in drac_client and fetch raid_controllers in this method as we do with all_physical_disks...?

Reply:
If we fetch raid_controllers in this method then I need to change the existing code , where this function get invoked in  define_target_raid_config function. In this function we already fetched raid_controllers.
line no. 263 : logical_disks = define_controller_logical_disks(drac_client, raid_controller_ids) 
I think we can keep it as is.